server-spi/src/main/java/org/keycloak/models/credential/OTPCredentialModel.java

```diff
@@ -59,7 +59,10 @@ public static OTPCredentialModel createFromPolicy(RealmModel realm, String secre
     public static OTPCredentialModel createFromCredentialModel(CredentialModel credentialModel) {
         try {
             OTPCredentialData credentialData = JsonSerialization.readValue(credentialModel.getCredentialData(), OTPCredentialData.class);
-            OTPSecretData secretData = JsonSerialization.readValue(credentialModel.getSecretData(), OTPSecretData.class);
+            OTPSecretData secretData = null;
+            if (credentialModel.getSecretData() != null) {
```

Comment:
Since the credentialModel may have the secret removed for security reasons, I'm having to add this guard clause.

Reply:
> > @stelewis-redhat let's not expand the scope of this PR to make everything configurable. I question that is needed in general, and it will also grow the scope of this PR a lot. So that should be a follow-up if not. For now let's focus on getting this ready to be merged.
> 
> Sorry, I didn't want to blow up the scope of this PR. However, I think being able to configure essential settings is an integral part of such a feature. But I agree, the feature is usable even without support for explicit configuration. However this should IMHO be added soon for completeness.
> 
> Some general guidance on feature completeness might help in the future? If we are striving to get something working initially, but it's not production grade then we should put it behind a preview flag. When something is "production grade" has to be defined.

OK I think I have everything done, except for turning on the feature flag.